### PR TITLE
adding the save_test_result module.

### PR DESCRIPTION
**Pull request by shawn-hurley**
Tied to proposal https://github.com/fusor/ansible-playbook-bundle/pull/92

### CODE REVIEW

File: library/asb_save_test_result.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+     - Takes a test result object and makes it available during the testing
+     action.
+notes: []
+requirements: []
+author:
+    - "Red Hat, Inc."
+options:
+  fail:
+    description:
+      - 'Will signify whether the test result has passed or faild.'
```
Comment (jwmatthews) [inline library/asb_save_test_result.py]: s/faild/failed

File: library/asb_save_test_result.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    default: false
+    type: bool
+  msg:
+    description:
+      - 'Determines the message th should be displayed for the failure or
```
Comment (jwmatthews) [inline library/asb_save_test_result.py]: 'th' seems strange

File: library/asb_save_test_result.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            if ansible_module.params['fail']:
+                test_result_file.write("1\n")
+            else:
+                # If failue is false, then 0 code signals success.
+                test_result_file.write("0\n")
+
+            if ansible_module.params['msg'] is not None:
+                test_result_file.write(ansible_module.params['msg'])
+    except Exception as error:
+        ansible_module.fail_json(msg="Error attempting to write binding: " + str(error))
```
Comment (cfchase) [inline library/asb_save_test_result.py]: s/binding/"test result"

File: library/asb_save_test_result.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+- name: test passed
+  asb_save_test_result:
+    msg: "Test passed!"
+'''
+RETURN = '''
+'''
+
+import json
+import base64
```
Comment (cfchase) [inline library/asb_save_test_result.py]: unnecessary import?

File: library/asb_save_test_result.py (hunk 2 above)
Comment (shawn-hurley) [reply library/asb_save_test_result.py]: Updated the wording completely

File: tests/test_asb_save_test_result.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    result_file_path: /var/tmp/test-result
+  roles:
+    - ansibleplaybookbundle.asb-modules
+  tasks:
+    - name: prepare test for clean run
+      file:
+        path: "{{ result_file_path }}"
+        state: absent
+
+    - name: encode bind credentials
```
Comment (cfchase) [inline tests/test_asb_save_test_result.yml]: might want to rename anywhere you use the word "bind"

File: tests/test_asb_save_test_result.yml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+      asb_save_test_result:
+        fail: true
+        msg: "Test failed"
+
```
Comment (cfchase) [inline tests/test_asb_save_test_result.yml]: Just wondering what the expected/actual behavior is if you do multiple test results.  Will it append new messages to the same file or will it overwrite the original file?  Seems like we'd want multiple test results to show in the file.

**Review by cfchase — CHANGES_REQUESTED** [review]

File: tests/test_asb_save_test_result.yml (hunk 6 above)
Comment (shawn-hurley) [reply tests/test_asb_save_test_result.yml]: I had thought that we would want to overwrite. But I think your right, that multiple results should be saved.

File: tests/test_asb_save_test_result.yml [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    - name: test result files should exist
+      debug:
+        var: result_file_status.stat.exists
+
+    - fail:
+        msg: Test result file at {{ result_file_path }} could not be found
+      when: not result_file_status.stat.exists
+
+    - name: read credentials file
```
Comment (cfchase) [inline tests/test_asb_save_test_result.yml]: You might want to change all the mentions of "credentials" or "encoded" in this file.  Other than that nitpick, the PR looks good to me.

**Review by cfchase — APPROVED** [review]

**Comment by shawn-hurley** [comment]
Wait to merge for https://github.com/fusor/ansible-playbook-bundle/pull/92 to merge in

**Comment by shawn-hurley** [comment]
[(13) Add ability for APB to test itself using a test action.](https://trello.com/c/D9l82yol/426-13-add-ability-for-apb-to-test-itself-using-a-test-action)

**Pull request merged**